src/lib/components/TableOfContents/index.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {html} from 'lit-element';
+import {BaseStateElement} from '../BaseStateElement';
+import {closeToC} from '../../actions';
+
+/**
+ * Element that renders table of contents.
+ * @extends {BaseStateElement}
+ * @final
+ */
+class TableOfContents extends BaseStateElement {
+  static get properties() {
+    return {
+      title: {type: String},
+      titleId: {type: String, attribute: 'title-id'},
+      opened: {type: Boolean, reflect: true},
+    };
+  }
+
+  constructor() {
+    super();
+    this.scrollSpy = this.scrollSpy.bind(this);
+    this.openedToFalse = this.openedToFalse.bind(this);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.tocHTML = this.innerHTML;
+    this.divContent =
```

Comment:
Can you pass #content in as a property or attribute?

Reply:
Suggest renaming to something like `articleContent`, as i'm not sure including `div` signifies much.

Also might want to log a warning if it's empty as I don't think that's a state we want to be in.